libs/community/langchain_community/document_loaders/parsers/language/php.py

```diff
@@ -0,0 +1,35 @@
+from typing import TYPE_CHECKING
+
+from langchain_community.document_loaders.parsers.language.tree_sitter_segmenter import (  # noqa: E501
+    TreeSitterSegmenter,
+)
+
+if TYPE_CHECKING:
+    from tree_sitter import Language
+
+
+CHUNK_QUERY = """
+    [
+        (function_definition) @function
+        (class_declaration) @class
+        (interface_declaration) @interface
+        (trait_declaration) @trait
+        (enum_declaration) @enum
+        (namespace_definition) @namespace
+    ]
+""".strip()
+
+
+class PHPSegmenter(TreeSitterSegmenter):
+    """Code segmenter for PHP."""
+
+    def get_language(self) -> "Language":
+        from tree_sitter_languages import get_language
```

Comment:
Would be nice to add a try/except here with a nice error message, similar to https://github.com/langchain-ai/langchain/blob/356a139b0a6c1ac5a0eff95c265ce226a91838bb/libs/community/langchain_community/document_transformers/html2text.py#L31

Reply:
ah see this is covered in TreeSitterSegmenter.